File: csharp/etl/Etl.cs
public static class Etl
{
    public static Dictionary<string, int> Transform(Dictionary<int, string[]> old)
    {
        Dictionary<string, int> transformed = [];

        foreach (var (occ, letters) in old)
        {
            foreach (var letter in letters)
            {
                transformed.Add(letter.ToLower(), occ);
            }
        }
        return transformed;
    }
}

File: kotlin/etl/src/main/kotlin/ETL.kt
object ETL {
    fun transform(source: Map<Int, Collection<Char>>): Map<Char, Int> {
        val result = mutableMapOf<Char, Int>()
        for ((occurencies, letters) in source.entries) {
            for (letter in letters) {
                result.put(letter.lowercaseChar(), occurencies)
            }
        }
        return result
    }
}
